File: .copilot/skills/workflow/examples/classify-route.cwf.py
tickets = args or ["app crashes on export", "please add dark mode", "how do I reset my password?"]
no_tools = wf.quarantine(allow_all_tools=False)


def classify_ticket(ticket):
    return wf.classify(
        ticket,
        ["bug", "feature", "question"],
        model="claude-haiku-4.5",
        label="classify",
        **no_tools,
    )


def suggest_action(kind, ticket):
    detail = wf.agent(
        f"In one sentence, suggest the next action for this {kind}: {ticket}",
        model="claude-haiku-4.5",
        phase="triage",
        label=kind,
        **no_tools,
    )
    return {"ticket": ticket, "kind": kind, "action": detail.content.strip()}


rows = [row for row in wf.pipeline(tickets, classify_ticket, suggest_action) if row is not None]
report = wf.synthesize(
    [f"{row['kind']}: {row['ticket']} -> {row['action']}" for row in rows],
    prompt="Group these by kind into a short triage report.",
    model="claude-sonnet-4.5",
    label="report",
    **no_tools,
)
print(report.content)

File: .copilot/skills/workflow/examples/classify-route.cwf.mjs
// classify-route.cwf.mjs — classify each ticket into a category, suggest an action, summarize.
export const meta = { name: "classify-route", description: "Classify tickets, suggest next actions, and synthesize a triage report." };

const tickets = args || ["app crashes on export", "please add dark mode", "how do I reset my password?"];
const noTools = quarantine({ allowAllTools: false });

const classifyTicket = (ticket) => classify(ticket, ["bug", "feature", "question"], { label: "classify", ...noTools });

const suggestAction = async (kind, ticket) => {
	const detail = await agent(`In one sentence, suggest the next action for this ${kind}: ${ticket}`, { agentType: "worker", phase: "triage", label: kind, ...noTools });
	return { ticket, kind, action: detail.content.trim() };
};

const rows = (await pipeline(tickets, classifyTicket, suggestAction)).filter((row) => row !== null);
const report = await synthesize(
	rows.map((row) => `${row.kind}: ${row.ticket} -> ${row.action}`),
	{ prompt: "Group these by kind into a short triage report.", label: "report", ...noTools },
);
return report.content;
